### PR TITLE
Add project view JSON endpoints [SCI-2728]

### PR DESCRIPTION
**Pull request by okriuchykhin**
https://biosistemika.atlassian.net/browse/SCI-2728

### CODE REVIEW

**Comment by okriuchykhin** [comment]
Most likely you have postgres 9.4 data files, while docker-compose.yml already has 9.6. Therefore postgres fails to start. You can clean up db container if you don't have any valuable data, or you can temporary change db version in config, than dump all data and import it to new db.

File: app/models/my_module.rb [hunk 1]
```diff
@@ -66,6 +66,13 @@ class MyModule < ApplicationRecord
   has_many :protocols, inverse_of: :my_module, dependent: :destroy
 
   scope :is_archived, ->(is_archived) { where('archived = ?', is_archived) }
+  scope :active, -> { where(archived: false) }
+  scope :overdue, -> { where('my_modules.due_date < ?', Time.current.utc) }
```
[the remark this answers is not in the document]
Comment (okriuchykhin) [reply app/models/my_module.rb]: Don't think that such precision is needed here

File: app/services/projects_overview_service.rb [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    {
+      '1' => 'projects.archived',
+      '2' => 'projects.name',
+      '3' => 'projects.created_at',
+      '4' => 'projects.visibility',
+      '5' => 'projects.user_count',
+      '6' => 'projects.experiment_count',
+      '7' => 'projects.task_count'
+    }
+  end
```
[the remark this answers is not in the document]
Comment (okriuchykhin) [reply app/services/projects_overview_service.rb]: I think DT uses strings as keys

File: app/controllers/my_modules_controller.rb [hunk 3]
```diff
@@ -382,7 +382,7 @@ def repository_index
                                              current_user,
                                              @my_module)
     @assigned_rows = records.assigned_rows
-    @repository_row_count = records.repository_rows.count
+    @repository_row_count = records.repository_rows.length
```
[the remark this answers is not in the document]
Comment (okriuchykhin) [reply app/controllers/my_modules_controller.rb]: Interesting, size should not run SQL COUNT if collection is loaded, but for this case it actually runs extra count, so explicit call of length works better

[the remark this answers is not in the document]
Comment (mz3944) [reply app/controllers/my_modules_controller.rb]: Strange... :+1:

File: app/services/projects_overview_service.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (mz3944) [reply app/services/projects_overview_service.rb]: I know, but  you can cast it and use it as `int` here. Doesn't really matter, forget it.

**Comment by mz3944** [comment]
Yes, I forgot how to completely wipe Docker volumes - I used `docker volume prune` to also remove the problematic `scinote-web_scinote_development_postgres` Docker volume, hence be able to pull the updated Postgres Docker image. Otherwise, I guess I would have to go en delete it manually.

**Pull request merged**